kolibri/plugins/management/assets/src/vuex/actions.js

```diff
@@ -10,93 +10,62 @@ function setSelectedGroupId({ dispatch }, id) {
-      );
-      xhr2.send();
-    }
-  };
-  xhr1.send();
+// An action for setting up the initial state of the app by fetching data from the server
+function fetch({ dispatch }) {
+  const learnerGroupPromise = new Promise((resolve) => {
+    classroomCollection.fetch().then(() => {
+      const cids = classroomCollection.map(c => c.id);
+      learnerGroupCollection.fetch({ params: { parent__id__in: cids } }).then((groupData) => {
+        resolve(groupData);
+      });
+    });
+  });
 
-  xhr3.open('GET', learnerListUrl);
-  xhr3.send();
+  const learnerPromise = learnerCollection.fetch();
 
-  xhr4.open('GET', membershipListUrl);
-  xhr4.send();
+  const memberPromise = memberCollection.fetch();
 
-  // This block of code is executed if xhr2, xhr3, and xhr4 all return with status code 200.
-  // xhr1 is not included because my simple promise wrapper sets the onreadystatechange property.
+  const promises = [learnerGroupPromise, learnerPromise, memberPromise];
+  // This block of code is executed if learnerGroup, learner, and membership
+  // fetching all return with status code 200.
+  // class fetching is not included because learnerGroup fetching depends upon it.
   Promise.all(promises).then(() => {
-    const learnerGroups = JSON.parse(xhr2.response);
-    const learners = JSON.parse(xhr3.response);
-    const memberships = JSON.parse(xhr4.response);
-
     const groupedLearners = new Set();
-    dispatch('ADD_LEARNER_GROUPS', learnerGroups.map(group => {
-      const learnerIds = memberships.filter(m => m.collection === group.id)
+    dispatch('ADD_LEARNER_GROUPS', learnerGroupCollection.map(group => {
+      const learnerIds = memberCollection.filter(m => m.collection === group.id)
         .map(m => m.user);
       learnerIds.forEach(id => groupedLearners.add(id));
       return Object.assign({}, group, {
-        learners: learners.filter(learner => learnerIds.indexOf(learner.id) !== -1)
+        learners: learnerCollection.filter(learner => learnerIds.indexOf(learner.id) !== -1)
           .map(learner => learner.id),
       });
     }));
 
     dispatch(
       'ADD_CLASSROOMS',
-      classrooms.map(classroom => {
-        const learnerIds = memberships.filter(m => m.collection === classroom.id)
+      classroomCollection.map(classroom => {
+        const learnerIds = memberCollection.filter(m => m.collection === classroom.id)
           .map(m => m.user);
         const ungroupedLearners = learnerIds.filter(id => !groupedLearners.has(id));
         return Object.assign(
           {},
           classroom,
           {
-            learnerGroups: learnerGroups.filter(g => g.parent === classroom.id)
+            learnerGroups: learnerGroupCollection.filter(g => g.parent === classroom.id)
               .map(g => g.id),
             ungroupedLearners,
           }
         );
       })
     );
-
-    dispatch('ADD_LEARNERS', learners);
+    dispatch('ADD_LEARNERS', learnerCollection.map(a => a));
```

Comment:
Apples to apples? Is this just to "de-collection-ify" it? If that's something we always need to do for this stage, perhaps we can abstract that out to something automatic or at least reusable.